src/polar/RadialBar.tsx

```diff
@@ -193,12 +194,12 @@ function SectorsWithAnimation({
 
         if (t > 0) {
           // eslint-disable-next-line no-param-reassign
-          previousSectorsRef.current = stepData;
+          previousSectorsRef.current = stepData ?? null;
         }
 
         return (
           <Layer>
-            <RadialBarSectors sectors={stepData} allOtherRadialBarProps={props} showLabels={!isAnimating} />
+            <RadialBarSectors sectors={stepData ?? []} allOtherRadialBarProps={props} showLabels={!isAnimating} />
```

Comment:
The empty arrays defined inline may cause extra re-renders (and infinite loops in the worst case). Try to return them from selectors (where they will be automatically memoized) or declare a const outside of the component so that it stays the same instance and React will not trigger a new render.